book.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'item'
+require 'json'
+
+class Book < Item
+  attr_reader :publisher, :cover_state
+
+  def initialize(publisher, cover_state, publish_date, title, id: SecureRandom.uuid)
+    super(publish_date)
+    @id = id
+    @publisher = publisher
+    @cover_state = cover_state
+    @title = title
+  end
+
+  def can_be_archived?
+    super || @cover_state == 'bad'
+  end
+
+  def to_json(*_args)
+    {
+      id: @id,
+      title: @title,
+      publish_date: @publish_date,
+      archived: @archived,
+      publisher: @publisher,
+      cover_state: @cover_state,
+      label_id: @label&.id
+    }.to_json
+  end
+
+  def to_s
+    "Book: title = #{@title}, publisher = #{@publisher}, " \
+      "cover_state = #{@cover_state}, publish_date = #{@publish_date}, " \
+      "archived = #{@archived}, label_title = #{@label&.title}, " \
+      "label_color = #{@label&.color}"
+  end
+  private :can_be_archived?
+end
```

Comment:
Hi @muneebulrehman, I added title property to the book class because the examples of the labels given are misleading to readers to think that the purpose of the label is to categorize the items, instead of identifying them individually. By the way, thank you very much for the review.
Ref: https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md#:~:text=List%20all%20labels%20(e.g.%20%27Gift%27%2C%20%27New%27)

Reply:
Yes, I understand. But that is how it is supposed to be. I hope you understood my point.

Reply:
@muneebulrehman , But then how a label could have multiple items?